File: VolleyballAPI/VolleyballAPI/Entities/Training.cs
using System.ComponentModel.DataAnnotations;
using VolleyballAPI.Enums;
using VolleyballAPI.JoinTableTypes;

namespace VolleyballAPI.Entities
{
    public class Training
  {
    [Required]
    [Key]
    public Guid Id { get; set; }
    public Guid TeamId { get; set; }
    public Guid LocationId { get; set; }
    public Guid CoachId { get; set; }
    public DateTime Date { get; set; }
    public string Description { get; set; }
    public PriceType AcceptableTickets { get; set; }
    public string PictureLink { get; set; }
    public virtual Team Team { get; set; } 
    public virtual Location Location { get; set; }
    public virtual User Coach{ get; set; }
    public virtual ICollection<TrainingParticipant> Players { get; set; } 
    public virtual ICollection<FavouriteTraining> UserHasAsFavourite { get; set; } 

  }
}

File: VolleyballManagementAppFrontend/src/app/models/training.ts
import { PriceType } from './priceType';
import { Team } from './team';
import { User } from './user';
import { Location } from './location';

export interface Training {
  id: string;
  players: User[];
  userHasAsFavourite: User[];
  location: Location;
  date: Date;
  description: string;
  team: Team;
  coach: User;
  pictureLink: string;
  priceType: PriceType[];
}